fe/fe-core/src/main/java/org/apache/doris/rpc/BackendServiceProxy.java

```diff
@@ -59,23 +61,19 @@ public static BackendServiceProxy getInstance() {
     }
 
     public void removeProxy(TNetworkAddress address) {
-        lock.writeLock().lock();
-        try {
-            serviceMap.remove(address);
-        } finally {
-            lock.writeLock().unlock();
+        LOG.warn("begin to remove proxy: {}", address);
+        BackendServiceClient service;
+        synchronized (this) {
```

Comment:
why use synchronized here and use lock for getProxy? how about use ConcurrentHashMap and use computeIfAbsent method

Reply:
fixed.
but `computeIfAbsent` is not suitable because we need to make sure that only one thread can create client for a certain BE.